core/services/registrysyncer/orm.go

```diff
@@ -0,0 +1,277 @@
+package registrysyncer
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/smartcontractkit/libocr/ragep2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+type capabilitiesRegistryDON struct {
```

Comment:
With the new types we've introduced inside the registrysyncer, you can go ahead and just use those types directly rather than having different types for the DB storage (there's still one type that uses the generated type and I would copy that here for now to allow further changes)

Reply:
I don't think I can just use the types since they can't be 1:1 into the DB without marshalling, which is why I have the other types. Examples of fields that will need to change type before hitting the DB are:

- DON.capabilities.DON.Members
- LocalRegistry.IDsToNodes[PeerID -> string].HashedCapabilityIds
- LocalRegistry.IDsToNodes[PeerID -> string].CapabilitiesDONIds
- LocalRegistry.IDsToNodes[PeerID -> string].Signer
- LocalRegistry.IDsToNodes[PeerID -> string].P2pId

So taking into consideration that the types in the `registrysyncer` could be expanded and we could end up with more fields having to be manually parsed/marshalled, I would prefer keeping the DB types separate.

@cedric-cordenier

Reply:
@vyzaldysanchez Is PeerID the only problem? PeerID implements MarshalText which MarshalJSON will use as a fallback so I don't think that would be a problem.

See: https://pkg.go.dev/encoding/json#Marshal